### PR TITLE
Add docs for ACK default tags

### PR DESCRIPTION
**Pull request by vijtrip2**
Description of changes:
* Add docs for ACK default tags

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/content/docs/user-docs/ack-tags.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+weight: 67
+toc: true
+---
+
+ACK controllers add default tags to AWS resources that support tagging.
+These default tags help in identifying and categorizing the AWS resources
+managed by ACK controllers.
+
+The two default tags added by ACK controller are `services.k8s.aws/controller-version`
+and `services.k8s.aws/namespace`. *controller-version* tag value is the name of
```
Comment (RedbackThomson) [inline docs/content/docs/user-docs/ack-tags.md:18]:
```suggestion
and `services.k8s.aws/namespace`. The *controller-version* tag value is the name of
```

File: docs/content/docs/user-docs/ack-tags.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+corresponding AWS service and version for that controller(Ex: s3-v0.1.3).
+And the *namespace* tag value is the namespace for the ACK resource.(Ex: default)
+
+When tags are already present inside the resource spec, ACK default tags are
+added in AWS resource alongside the tags from resource spec. Priority is given to
+resource spec tags when there is a conflict between ACK default tag keys and tag
+keys in resource spec.
+
+Example:
```
Comment (RedbackThomson) [inline docs/content/docs/user-docs/ack-tags.md:27]: nit: Make into a header
```suggestion
### Example
```

File: docs/content/docs/user-docs/ack-tags.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+For a resource manifest like
+
+```yaml
+apiVersion: s3.services.k8s.aws/v1alpha1
+kind: Bucket
+metadata:
+  name: my-ack-tagging-bucket
```
Comment (RedbackThomson) [inline docs/content/docs/user-docs/ack-tags.md:35]: Please add `namespace: default` so people can directly see that the tag value matches the metadata value

File: docs/content/docs/user-docs/ack-tags.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+The default tags added by ACK controllers are configurable during controller
+installation.
+
+To remove the ACK default tags, include `--set 'resourceTags={}'` during helm
+chart installation
+
+To provide different value for default ACK tags, include
+`--set 'resourceTags={tk1=tv1, tk2=tv2}'` during helm chart installation.
+
+
```
Comment (RedbackThomson) [inline docs/content/docs/user-docs/ack-tags.md:85]: Can we also call out which variable substitutions we allow?

File: docs/content/docs/user-docs/ack-tags.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
+---
+title: "Managing ACK Default Tags"
```
Comment (jaypipes) [inline docs/content/docs/user-docs/ack-tags.md:2]: I'm tempted to make this more generic... something like "Managing Tags on your AWS Resources"

File: docs/content/docs/user-docs/ack-tags.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+menu:
+  docs:
+    parent: "getting-started"
+weight: 67
+toc: true
+---
+
+ACK controllers add default tags to AWS resources that support tagging.
+These default tags help in identifying and categorizing the AWS resources
+managed by ACK controllers.
```
Comment (jaypipes) [inline docs/content/docs/user-docs/ack-tags.md:15]: and then before this paragraph, say something to the effect of:

> Most AWS resources can have one or more Tags -- simple key/value pairs -- associated with them. These Tags allow you to organize and categorize your AWS resources for accounting and informational purposes. ACK custom resources (CRs) that support Tags will have a `Spec.Tags` field that store user-defined key/value pairs. In addition to user-defined Tags, ACK also supports a set of *default tags*, which are Tags that the ACK controller will automatically ensure are on all resources that it manages.

File: docs/content/docs/user-docs/ack-tags.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+metadata:
+  name: my-ack-tagging-bucket
+spec:
+  name: my-ack-tagging-bucket
+  tagging:
+    tagSet:
+    - key: "first"
+      value: "1"
+    - key: "second"
+      value: "2"
```
[the remark this answers is not in the document]
Comment (jaypipes) [reply docs/content/docs/user-docs/ack-tags.md:43]: Completely agree. :) Let's use one of the CRDs that has a `Tags` field that is either a `map[string]string` or `[]struct{}` type.

**Review by jaypipes — CHANGES_REQUESTED** [review]
I'd like to see this actually be a tad more generic (see suggestions inline..)

File: docs/content/docs/user-docs/ack-tags.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+ACK also supports a set of default tags, which are Tags that the ACK
+controller will automatically ensure are on all resources that it manages.
+
+The two default tags added by ACK controller are `services.k8s.aws/controller-version`
+and `services.k8s.aws/namespace`. The *controller-version* tag value is the name of
+corresponding AWS service and version for that controller(Ex: s3-v0.1.3).
+And the *namespace* tag value is the namespace for the ACK resource.(Ex: default)
+
+When tags are already present inside the resource spec, ACK default tags are
+added in AWS resource alongside the tags from resource spec. Priority is given to
```
Comment (jaypipes) [inline docs/content/docs/user-docs/ack-tags.md:27]:
```suggestion
When tags are already present inside the Kubernetes custom resource's `Spec.Tags`, ACK default tags are
added to the AWS resource's tags collection along with those tags from `Spec.Tags`. Priority is given to
```

File: docs/content/docs/user-docs/ack-tags.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+resource spec tags when there is a conflict between ACK default tag keys and tag
```
Comment (jaypipes) [inline docs/content/docs/user-docs/ack-tags.md:28]:
```suggestion
`Spec.Tags` when there is a conflict between ACK default tag keys and tag
```

File: docs/content/docs/user-docs/ack-tags.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+keys in resource spec.
```
Comment (jaypipes) [inline docs/content/docs/user-docs/ack-tags.md:29]:
```suggestion
keys in `Spec.Tags`.
```

File: docs/content/docs/user-docs/ack-tags.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+  during helm chart installation.
+
+* ACK supports variable expansion inside tag values for following variables:
+  - %CONTROLLER_SERVICE%
+  - %CONTROLLER_VERSION%
+  - %K8S_NAMESPACE%
+  - %K8S_RESOURCE_NAME%
+
+  A custom resource tag `k8s-name=%K8S_RESOURCE_NAME` in above ecr repository example
+  will be expanded as "k8s-name=my-ack-tagging-repo"
```
Comment (jaypipes) [inline docs/content/docs/user-docs/ack-tags.md:102]:
```suggestion
  would be expanded to "k8s-name=my-ack-tagging-repo"
```

**Review by jaypipes — CHANGES_REQUESTED** [review]
Few minor grammar/nits otherwise looking great Vijay!

**Review by jaypipes — APPROVED** [review]
sweeeeeeeet. :)

**Pull request merged**